metabase/metabase
# [refs] `lib.convert` preserves `:ident`s through a round trip

**NOTE:** This PR got reverted by #51128 and the second edition is #51133.


Closes #49682. Closes #49683.

### Description

Updates `lib.convert` and legacy MBQL normalization to allow
`:ident`s, and fixes some QP middleware and utils so they can
still match up refs correctly.

No user-visible impact.

Newly created queries will have `:ident`s generated for joins,
aggregations, etc. by the previous query, and this will start
saving them into appDBs.

Pre-existing clauses don't have refs added yet.

### How to verify

Newly created "columnar" clauses (joins, aggregations, breakouts, expressions)
get `:ident`s generated by the lib, and these are now visible in legacy MBQL
and on the wire.

If you save a question, the generated `:ident`s will be returned when we read the
card again.

See the unit test changes for lots of examples of what this changed in legacy MBQL.

### Checklist
